modules/logger/logger.py

```diff
@@ -0,0 +1,135 @@
+"""
+Logs debug messages.
+"""
+
+import datetime
+import inspect
+import logging
+import pathlib
+import os
+import types
+import typing
+import yaml
+
+CONFIG_FILE_PATH = pathlib.Path("config.yaml")
+
+
+class Logger:
+    """
+    Instantiates Logger objects.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(cls, name: str) -> "tuple[bool, Logger | None]":
+        """
+        Create and configure a logger.
+        """
+
+        # Open config file.
+        try:
+            with CONFIG_FILE_PATH.open("r", encoding="utf8") as file:
+                try:
+                    config = yaml.safe_load(file)
+                except yaml.YAMLError as exc:
+                    print(f"Error parsing YAML file: {exc}")
+                    return -1
+        except FileNotFoundError:
+            print(f"File not found: {CONFIG_FILE_PATH}")
+            return False, None
+        except IOError as exc:
+            print(f"Error when opening file: {exc}")
+            return False, None
+
+        # Get the path to the logs directory.
+        log_directory_path = config["log_directory_path"]
+        entries = os.listdir(log_directory_path)
+        log_names = [
+            entry for entry in entries if os.path.isdir(os.path.join(log_directory_path, entry))
+        ]
+
+        # Find the log directory for the current run, which is the most recent timestamp
+        datetime_format = "%Y-%m-%d_%H-%M-%S"
+        log_path = max(
+            log_names,
+            key=lambda datetime_string: datetime.datetime.strptime(
+                datetime_string, datetime_format
+            ),
+        )
+
+        filename = f"{log_directory_path}/{log_path}/{name}.log"
+
+        # Formatting configurations for the logger
```

Comment:
Nit: I see some comments with periods and some that don't have periods. Can we try to make all the comments consistent?